src/main/java/com/whatsapp/api/WhatsappApiServiceGenerator.java

```diff
@@ -37,6 +44,46 @@ public class WhatsappApiServiceGenerator {
                 .build();
     }
 
+    /**
+     * Sets http proxy for the shared client.
+     * <p>If you need to use a proxy to connect to the internet,
+     * you can use this method to set it.
+     * <p>
+     * <ul>
+     * <li>If you want to use a proxy that requires authentication,
+     * you can pass the username and password as parameters.
+     * <p>
+     * <li>If you want to use a proxy that does not require authentication,
+     * you can pass {@code null} as parameters for {@code username} and {@code pwd}.
+     * </ul>
+     * <p>
+     * @param host     the host (Not null)
+     * @param port     the port (Not null)
+     * @param username the username
+     * @param pwd      the pwd
+     * @see <a href="https://square.github.io/okhttp/4.x/okhttp/okhttp3/-ok-http-client/-builder/proxy-selector/">Proxy Selector</a>
+     * @see <a href="https://square.github.io/okhttp/4.x/okhttp/okhttp3/-ok-http-client/-builder/proxy-authenticator/">Proxy Authenticator</a>
+     */
+    public static void setHttpProxy(String host, int port, String username, String pwd) {
+        Objects.requireNonNull(host, "Host cannot be null");
+        Objects.requireNonNull(port, "Http Port cannot be null");
```

Comment:
This control can be avoided as port is int and not Integer

Reply:
Thanks @fboragine!